### PR TITLE
TestLeaseAndUpdateTaskList_Sticky, TestGetDecisionTasks, TestCompleteDecisionTask, TestReplicationTasks, TestTimerTasks

### PR DESCRIPTION
**Pull request by uber-qlam**

### CODE REVIEW

File: common/persistence/sql/schema/down.sql [hunk 1]
```diff
@@ -4,5 +4,7 @@ DROP TABLE IF EXISTS executions;
 DROP TABLE IF EXISTS current_executions;
 DROP TABLE IF EXISTS shards;
 DROP TABLE IF EXISTS transfer_tasks;
+DROP TABLE IF EXISTS replication_tasks;
+DROP TABLE IF EXISTS timer_tasks;
 DROP TABLE IF EXISTS tasks;
 DROP TABLE IF EXISTS task_lists;
```
Comment (meiliang86) [inline common/persistence/sql/schema/down.sql:10]: New line at the end of file.

File: common/persistence/sql/schema/up.sql [hunk 2]
```diff
@@ -145,7 +145,7 @@ CREATE TABLE IF NOT EXISTS current_executions(
   PRIMARY KEY (shard_id, domain_id, workflow_id)
 );
 
-CREATE TABLE tasks (
+CREATE TABLE IF NOT EXISTS tasks (
```
Comment (meiliang86) [inline common/persistence/sql/schema/up.sql:148]: Let's be consistent on whether to do if not exists here. IMO if you always drop tables at the end of test then we should not have this here.

File: common/persistence/sql/sqlTaskPersistence.go [hunk 3]
```diff
@@ -161,7 +175,7 @@ func (m *sqlTaskManager) LeaseTaskList(request *persistence.LeaseTaskListRequest
 		}
 
 		if result, err := tx.NamedExec(updateTaskListSQLQuery,
-			struct{
+			struct {
```
Comment (meiliang86) [inline common/persistence/sql/sqlTaskPersistence.go:178]: Can you define these along with others in the definition section? Call it "updateTaskListRow" maybe.

File: common/persistence/sql/sqlTaskPersistence.go [hunk 4]
```diff
@@ -124,7 +138,7 @@ func (m *sqlTaskManager) LeaseTaskList(request *persistence.LeaseTaskListRequest
 			if _, err := m.db.NamedExec(createTaskListSQLQuery,
 				tasksListsRow{
```
Comment (meiliang86) [inline common/persistence/sql/sqlTaskPersistence.go:139]: tasksListsRow -> tasksListRow

File: common/persistence/sql/sqlTaskPersistence.go [hunk 5]
```diff
@@ -212,61 +226,78 @@ func (m *sqlTaskManager) LeaseTaskList(request *persistence.LeaseTaskListRequest
...
-
-	if result, err := tx.NamedExec(updateTaskListSQLQuery,
-		struct {
-		tasksListsRow
-		OldRangeID int64 `db:"old_range_id"`
-	}{
-		tasksListsRow{
+	if request.TaskListInfo.Kind == persistence.TaskListKindSticky {
+		// If sticky, update with TTL
+		if _, err := m.db.NamedExec(updateTaskListWithTTLSQLQuery, tasksListsRow{
```
Comment (meiliang86) [inline common/persistence/sql/sqlTaskPersistence.go:231]: I believe you can use pointer for the second param.

**Comment by uber-qlam** [comment]
abandoned due to rebase issues. see 
https://github.com/uber-qlam/cadence/pull/7/

**Pull request closed without merging**